api/courses.go

```diff
@@ -353,21 +353,42 @@ type renameLectureRequest struct {
 	Name string
 }
 
-func deleteLecture(c *gin.Context) {
+func deleteLectures(c *gin.Context) {
 	foundContext, exists := c.Get("TUMLiveContext")
 	if !exists {
 		sentry.CaptureException(errors.New("context should exist but doesn't"))
 		c.AbortWithStatus(http.StatusInternalServerError)
 		return
 	}
 	tumLiveContext := foundContext.(tools.TUMLiveContext)
-	stream, err := dao.GetStreamByID(context.Background(), c.Param("streamID"))
-	if err != nil || stream.CourseID != tumLiveContext.Course.ID {
-		c.AbortWithStatus(http.StatusForbidden)
+
+	jsonData, err := ioutil.ReadAll(c.Request.Body)
+	if err != nil {
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	var req deleteLecturesRequest
+	err = json.Unmarshal(jsonData, &req)
+	if err != nil {
+		c.AbortWithStatus(http.StatusBadRequest)
 		return
 	}
-	stream.Model.DeletedAt = gorm.DeletedAt{Time: time.Now()} // todo ?!
-	dao.DeleteStream(strconv.Itoa(int(stream.ID)))
+
+	var streams []model.Stream
+	for _, streamID := range req.StreamIDs {
+		stream, err := dao.GetStreamByID(context.Background(), streamID)
+		if err != nil || stream.CourseID != tumLiveContext.Course.ID {
+			c.AbortWithStatus(http.StatusForbidden)
+			return
+		}
+		streams = append(streams, stream)
+	}
+
+	for _, stream := range streams {
+		stream.Model.DeletedAt = gorm.DeletedAt{Time: time.Now()} // todo ?!
```

Comment:
You don't need to set this, that's what gorm internally does when you call `Delete()`

Reply:
Nice thanks, I will make the changes ASAP! :)

Reply:
What do you think now ? :)
![Bildschirmfoto 2021-12-31 um 16 48 37](https://user-images.githubusercontent.com/17506411/147831293-a8e51184-2721-4f68-ab8e-fae2a2f22711.png)